File: rpg/schemas/permissions.py
"""schemas.permissions — 权限/确认管理路由请求模型。"""
from __future__ import annotations
from typing import Optional, Any
from schemas._common import _BaseRequest


class PermissionsRequest(_BaseRequest):
    mode: Optional[str] = "full_access"


class PendingWriteRequest(_BaseRequest):
    id: Optional[Any] = None
    index: Optional[Any] = None
    action: Optional[str] = None
    decision: Optional[str] = None


class QuestionClearRequest(_BaseRequest):
    id: Optional[Any] = None
    index: Optional[Any] = None
    choice: Optional[Any] = None


class DebugPendingQuestionRequest(_BaseRequest):
    text: Optional[str] = None

File: rust/crates/rpg-schemas/src/permissions.rs
//! schemas.permissions — 权限/确认管理路由请求模型。
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PermissionsRequest {
    #[serde(default = "default_perm_mode")]
    pub mode: Option<String>,
}

fn default_perm_mode() -> Option<String> { Some("full_access".to_string()) }

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct PendingWriteRequest {
    pub id: Option<Value>,
    pub index: Option<Value>,
    pub action: Option<String>,
    pub decision: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct QuestionClearRequest {
    pub id: Option<Value>,
    pub index: Option<Value>,
    pub choice: Option<Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct DebugPendingQuestionRequest {
    pub text: Option<String>,
}
